session/create_consumer.go

```diff
@@ -70,14 +80,33 @@ func (consumer *createConsumer) Consume(requestPtr interface{}) (response interf
 				destroyCallback()
 			}()
 		}
-		return responseWithSession(sessionInstance, config, nil), nil
+		return responseWithSession(sessionInstance, config, consumer.loadPaymentInfo(issuerID)), nil
 	case ErrorInvalidProposal:
 		return responseInvalidProposal, nil
 	default:
 		return responseInternalError, nil
 	}
 }
 
+func (consumer *createConsumer) loadPaymentInfo(issuerID identity.Identity) *PaymentInfo {
```

Comment:
You are testing private method in this case. Imho it should be tested, but maybe it can go to some payment related package and made public? Promise is cross session persistent. It is not something unique to a single session.